deis/builder
# ref(Makefile): use gen-changelog.sh from go-dev

[skip ci]

# Summary of Changes

Replaces the local `_scripts/generate-changelog.sh` with the improved new standard version `gen-changelog.sh` in the `quay.io/deis/go-dev:0.12.0` image.

If this approach is acceptable, a slew of copycat PRs are to follow. So I would appreciate feedback on whether `make update-changelog` is a good target name, if this could be further simplified, and whether it automates enough of the Deis Worfklow release process before I forge ahead.

cc: @bacongobbler @vdice

# Issue(s) that this PR References

deis/docker-go-dev#35.

# Testing Instructions

1. Run `make update-changelog`
2. Run `git diff` and examine the output
3. Open `CHANGELOG.md` in an editor

Expected Results:

1. `CHANGELOG.md` should have a new section at the top
2. The section header should reference the latest tag in your local builder codebase
3. The section header should reference `HEAD` (or `$DEIS_RELEASE` if that env var is set)
4. The updated `CHANGELOG.md` should render correctly as Markdown
5. Links in the updated `CHANGELOG.md` should work

# Pull Request Hygiene TODOs

Please make sure the below checklist is complete.
